gnomad_constraint/utils/constraint.py

```diff
@@ -396,6 +406,183 @@ def apply_models(
+    max_cov: int = 60,
+    gerp_lower_cutoff: float = -3.9885,
+    gerp_upper_cutoff: float = 2.6607,
+) -> hl.Table:
+    """
+    Calculate mutation rate using the downsampling with size specified by `downsampling_level` in genome sites Table.
+
+    Prior to computing mutation rate the only following variants are kept:
+        - variants with the mean coverage in the gnomAD genomes between `min_cov` and `max_cov`.
+        - variants where the most severe consequence was 'intron_variant' or
+            'intergenic_variant'.
+        - variants with the GERP score between `gerp_lower_cutoff` and `gerp_upper_cutoff` (these default to -3.9885 and
+        2.6607, respectively - these values were precalculated on the GRCh37 context
+        Table and define the 5th and 95th percentiles).
+        - high-quality variants: `exome_ht.pass_filters`.
+        - variants with allele count below `ac_cutoff`: `(freq_expr.AC <= ac_cutoff)`.
+
+    The returned Table includes the following annotations:
+        - context - trinucleotide genomic context.
+        - ref - the reference allele.
+        - alt - the alternate base.
+        - methylation_level - methylation_level.
+        - downsampling_counts_{pop} - variant counts in downsamplings for populations
+          in `pops`.
+        - mu_snp - SNP mutation rate.
+        - annotations added by `annotate_mutation_type`.
+
+    :param genome_ht: Genome sites Table for autosome/pseudoautosomal regions.
+    :param context_ht: Context Table for autosome/pseudoautosomal regions.
+    :param recalculate_all_possible_summary: Whether to calculate possible variants
+        using context Table with locus that is only on an autosome or in a
+        pseudoautosomal region. Default is True.
+    :param omit_methylation: Whether to omit 'methylation_level' from the grouping when
+        counting variants. Default is False.
+    :param count_singletons: Whether to count singletons. Default is False.
+    :param keep_annotations: Annotations to keep in the context Table and genome sites
+        Table.
+    :param ac_cutoff: The cutoff of allele count when filtering context Table and genome sites Table.
+    :param downsampling_level: The size of downsamplings will be used to count variants. Default is 1000.
+    :param total_mu: The per-generation mutation rate. Default is 1.2e-08.
+    :param pops: List of populations to use for downsampling counts. If empty Tuple is supplied, will default to '['global']'.
+    :param min_cov: Minimum coverage required to keep a site when calculating the mutation rate. Default is 15.
+    :param max_cov: Maximum coverage required to keep a site when calculating the mutation rate. Default is 60.
+    :param gerp_lower_cutoff: Minimum gerp score for variant to be included when calculating the mutation rate. Default is -3.9885.
+    :param gerp_upper_cutoff: Maximum gerp score for variant to be included when calculating the mutation rate. Default is 2.6607.
```

Comment:
reformat so they don't go over the 80 char lines

Reply:
do you use any particular tools to automate this?

Reply:
unfortunately no, avery and I tried one, but it didn't do a good job at all, so we abandoned it